tests/mmap_cache.rs

```diff
@@ -0,0 +1,108 @@
+
+use common::*;
+
+/// Check that an invalid cache results in the process exiting.
+/// We expect the process here to exit(1).
+/// I don't think we can effectively test the exit code in a test
+/// so I'm just ignoreing it.
+#[ignore]
+#[tokio::test]
+async fn mmap_cache() -> eyre::Result<()> {
+    let _ = tracing_subscriber::fmt::try_init();
+
+    let mut cache_file = NamedTempFile::new()?;
+    // Write an invalid cache file
+    cache_file.write(&[
+        // length
+        4, 0, 0, 0, 0, 0, 0, 0, //
+        // num leaves
+        2, 0, 0, 0, 0, 0, 0, 0, //
+        0, 0, 0, 0, 0, 0, 0, 0, //
+        0, 0, 0, 0, 0, 0, 0, 0, //
+        0, 0, 0, 0, 0, 0, 0, 0, //
+        // 1
+        0, 0, 0, 0, 0, 0, 0, 0, //
+        0, 0, 0, 0, 0, 0, 0, 0, //
+        0, 0, 0, 0, 0, 0, 0, 0, //
+        0, 0, 0, 0, 0, 0, 0, 0, //
+        // 2
+        0, 0, 0, 0, 0, 0, 0, 0, //
+        0, 0, 0, 0, 0, 0, 0, 0, //
+        0, 0, 0, 0, 0, 0, 0, 0, //
+        0, 0, 0, 0, 0, 0, 0, 0, //
+        // 3
+        0, 0, 0, 0, 0, 0, 0, 0, //
+        0, 0, 0, 0, 0, 0, 0, 0, //
+        0, 0, 0, 0, 0, 0, 0, 0, //
+        0, 0, 0, 0, 0, 0, 0, 0, //
+    ])?;
+
+    let mainnet_container = setup_mainnet().await?;
+    let mainnet_rpc_port = mainnet_container.get_host_port_ipv4(8545).await?;
+    let mainnet_rpc_url = format!("http://127.0.0.1:{mainnet_rpc_port}");
+
+    let rollup_container = setup_rollup().await?;
+    let rollup_rpc_port = rollup_container.get_host_port_ipv4(8545).await?;
+    let rollup_rpc_url = format!("http://127.0.0.1:{rollup_rpc_port}");
+
+    let tree = CascadingMerkleTree::<PoseidonHash, _>::new(
+        vec![],
+        TREE_DEPTH,
+        &Field::ZERO,
+    );
+
+    let initial_root = tree.root();
+    tracing::info!(?initial_root, "Initial root",);
+
+    let id_manager_address: Address =
+        "0x5FbDB2315678afecb367f032d93F642f64180aa3".parse()?;
+    let bridged_address: Address =
+        "0x5FbDB2315678afecb367f032d93F642f64180aa3".parse()?;
+
+    tracing::info!("Setting up world-tree service");
+
+    let service_config = ServiceConfig {
+        tree_depth: TREE_DEPTH,
+        canonical_tree: TreeConfig {
+            address: id_manager_address,
+            window_size: 10,
+            creation_block: 0,
+            provider: ProviderConfig {
+                rpc_endpoint: mainnet_rpc_url.parse()?,
+                throttle: 150,
+            },
+        },
+        cache: CacheConfig {
+            cache_file: cache_file.path().to_path_buf(),
+            purge_cache: false,
+        },
+        bridged_trees: vec![TreeConfig {
+            address: bridged_address,
+            window_size: 10,
+            creation_block: 0,
+            provider: ProviderConfig {
+                rpc_endpoint: rollup_rpc_url.parse()?,
+                throttle: 150,
+            },
+        }],
+        socket_address: None,
+        telemetry: None,
+    };
+
+    let (_, _) = setup_world_tree(&service_config).await?;
```

Comment:
Instead of `process::exit`, we could panic and then try grabbing all handles here and asserting that at least one of them failed with an error. We would have to push the blocking handle through the stack

Reply:
Yeah we have sort of three options here:
- set a panic handle
- exit
- send a message through a channel to the main thread.

You have a strong preference?

Reply:
Is there a reason you opted to use `process::exit(1)`? I lean towards panic if the tree state is corrupted as it can be unwound (panicking also feels clean here).

Reply:
> Instead of `process::exit`, we could panic and then try grabbing all handles here and asserting that at least one of them failed with an error. We would have to push the blocking handle through the stack

I think I prefer directly exiting here. Is there any reason you'd want to panic instead?

Reply:
Let's stick to exiting